app/views/bed_layouts/_destroy_location.html.haml

```diff
@@ -0,0 +1,31 @@
+%div{data: { barcodes: barcodes || "", location: location }}
+  %p{style: "background-color: #e8f5e9;color: green;padding: 10px; width: 80%"}
+    Click the Search button to view all labware available in the selected location.
+%div
+  %label{for: "location"} Source location
+  = text_field_tag 'location', location, :size => 80, :onkeydown => "if(event.key == 'Enter') { event.preventDefault(); window.dispatchSearchEvent();}"
+  = button_tag "Search", :type => "button", :id => "search_button", :onclick => "window.dispatchSearchEvent()"
+- if barcodes.present?
+  %div{style: "margin-top: 40px; text-align: left;"}
+    %p{style: "margin-left: 40px;color: green;background-color: #e8f5e9;padding: 10px; width: 80%"}
+      = "When you click the Submit button, the following #{barcodes.split(/\n/).length} labware items will be moved to the destroyed location in LabWhere"
+  %div
+    %label{for: "robot"} Source labware
+    = text_area_tag "robot", barcodes, :rows => 25, :cols => 80, :readonly => true
+- else
+  - flash.each do |flash_type, messages|
+    = content_tag(:div, messages, class: 'flash', id: "message_#{flash_type}", style: "margin-top: 20px;width: 60%")
+:javascript
+  window.dispatchSearchEvent = function() {
+    $.post('#{pre_validate_layout_bed_layouts_path}',{
+            instrument_barcode: $('#instrument_barcode').val(),
+            instrument_process_id: $('#instrument_process').val(),
+            scanned_values: $('#location').val()
```

Comment:
I found the answer for a previous question here. I just want to point out all other bed verification classes use the `robot` param to hold the large text field value where labware barcodes are scanned. The PlatesController creates an instance of bed verification class and passes params['robot'] as scanned_values argument to initialisation and it is available as attributes['scanned_values'] in the verification instances. There are changes around this, e.g., introduction of scanned_values into controller params in bed_layout_verification, which made following the code difficult for me.

Reply:
In this case, we have two values: the location and the labware inside it. I didn't use the 'robot' field for the location, as I intend to use it for the fetched labware inside a location. This way, the ProcessPlatesController requires no changes, and it functions exactly the same way as it does for others.
Hope it is ok. What I will do is, I will  try to put in more comments for scanned_values in bed_layout_verification.